.github/workflows/ci.yml

```diff
@@ -350,3 +350,39 @@ jobs:
           emmake make -C build qjs_wasm -j$(getconf _NPROCESSORS_ONLN)
       - name: result
         run: ls -lh build
+
+  cygwin:
+    runs-on: windows-latest
+    steps:
+      - name: Set up Cygwin
+        uses: cygwin/cygwin-install-action@master
+        with:
+            packages: make cmake gcc-g++ bash
+
+      - uses: actions/checkout@v3
+
+      - name: build
+        shell: C:\cygwin\bin\bash.exe --login --norc -eo pipefail -o igncr '{0}'
```

Comment:
You can put the shell in a default block to avoid having to repeat it.

Reply:
The shell is installed as part of the workflow, so I didn't think that would work. I'll try it, though.